The user ought to know where they are currently

gotonode opened the pull request:
Implements #945.

gotonode commented:
I'll implement changes to the Labtool logo and add the vertical bar to distinguish it from the other navigation elements. I'll also change the rounded corners on the navigation bar, so it'll look better.

Someone else take care of changing the navigation buttons based on where the user currently is?

gotonode commented:
Assign yourself to this pull request and to the related issue if you want to tackle changing the navigation buttons somehow, to indicate on what page the user is currently on.

gotonode commented:
Labtool 2.0 logo is now a button that redirects to "My page".

I tried messing around with Semantic UI's Divider-elements, but couldn't get them to work properly. I still believe the logo should be distinguished from other nav elements, such as the buttons. It's a link for accessibility purposes, and because, why not?

mjaakko commented:
This needs planning: what to show as active in top menu when user is not at one of the pages listed there?

hisahi commented:
I added some highlighting to the current page on the navbar - this only covers "my page", "courses", "import" and "admin", however.

hisahi commented:
> 
> 
> This needs planning: what to show as active in top menu when user is not at one of the pages listed there?

This only really happens for the student if we are on a course page - it's probably fine to not highlight anything in that case

gotonode commented:
Maybe when the user is on page related to courses, highlight that selection in the main menu? Or perhaps highlight it in a different way? One way when the user is on the exact page, and another way (slightly different) when the user is a page related to it?

Furthermore, the Labtool 2.0 logo doesn't highlight all the way to the bottom for me on Chrome. It leaves a nasty rectangular shape in the original color of the main menu underneath it.

gotonode commented:
Also there's some jitter on the main menu since sometimes the Labtool 2.0 logo's text is in bold and sometimes it's not. It moves the navigation elements on the right of it ever so slightly.

gotonode commented:
I removed the rounded corners on the top of the main menu. I think it looks better this way. Another stylistic choice would've been to introduce a little margin on top of the main menu; in which case the rounded corners on the top would've made sense and looked better.

gotonode commented:
While we're at it (the main menu), should we add an icon to the Logout-button? Or is it fine the way it is? An icon like a man going through a door or something.

hisahi commented:
> 
> 
> Maybe when the user is on page related to courses, highlight that selection in the main menu? Or perhaps highlight it in a different way? One way when the user is on the exact page, and another way (slightly different) when the user is a page related to it?
> 
> Furthermore, the Labtool 2.0 logo doesn't highlight all the way to the bottom for me on Chrome. It leaves a nasty rectangular shape in the original color of the main menu underneath it.
I don't remember having this issue, but we could just remove the highlight from the logo



> 
> 
> While we're at it (the main menu), should we add an icon to the Logout-button? Or is it fine the way it is? An icon like a man going through a door or something.
Yeah, I would add some icon there

gotonode commented:
The Logout-button looks very dull and boring. Maybe we'll just add the icon, and consider changing its color a little bit?

gotonode commented:
Can we limit this issue's content so we can merge it? Or can we get this into production by Sunday?

gotonode commented:
Added Logout-button icon, changed its color. Feel free to use a better color if you can find one.

For accessibility purposes, I think the Logout-button should say the first name of the currently logged-in user.

For example: **Logout Riku**

gotonode commented:
Should we merge this PR before the final sprint is over? We could change the PR's scope so that it fulfills its requirements.

gotonode commented:
LGTM but the Labtool 2.0 logo still shows a nasty bar below it when active (on Chrome).

gotonode commented:
Hopefully this gets into production before the demo.

hisahi commented:
Wouldn't bolding be better than underlining? The logout button also looks quite menacing right now

yumoL commented:
The logout button is still visible although a user hasn't logged in. On the other hand, it isn't a problem as the fake login page won't be used in production?

mjaakko commented:
> The logout button is still visible although a user hasn't logged in. On the other hand, it isn't a problem as the fake login page won't be used in production?

This only happens when using fake login.

mjaakko commented:
> Wouldn't bolding be better than underlining? 

Currently it does both. Maybe it would be better to just bold it, but then it might not be clear enough that it is "active".

> The logout button also looks quite menacing right now

I agree, maybe we should just change it back to white?

hisahi commented:
> 
> 
> > Wouldn't bolding be better than underlining?
> 
> Currently it does both. Maybe it would be better to just bold it, but then it might not be clear enough that it is "active".
>
The underlining looks a bit bad. Bolding alone might already help, but there might be something else to change too - I can't immediately come up with anything, though

> 
> > The logout button also looks quite menacing right now
> 
> I agree, maybe we should just change it back to white?
Probably a good idea.

gotonode commented:
I'm an author so I cannot review and accept this but LGTM.

hisahi commented:
This also resolves #979, FWIW.

hisahi commented:
Does it look good now?

mjaakko commented on labtool2.0/src/components/pages/Nav.css:
```diff
@@ -1,3 +1,10 @@
+}
+.ui.inverted.pointing.menu .active.item::after {
+    display: none !important;
+}
```
Custom CSS class name (instead of Semantic UI) should be used to apply these styles, as otherwise all Semantic UI menus will have these styles.

gotonode replied:
Yeah, even if those are not currently in use. In the future they might be.
